### ISSUE TITLE
Metric Learning for Large Scale Image Classification: Generalizing to New Classes at near-zero cost

### ISSUE THREAD

**Issue by sweaterr** [sweaterr/readinglist#360]

**Comment by sweaterr** [comment]
In the last decade we have witnessed an explosion in the amount of images and videos that are digitally available, e.g, in broadcasting archives or social media sharing websites.

Only a small fraction of this data is consistently annotated and thus scalable methods are needed for annotation and retrieval to efficiently access this huge volume of data.

This need has been recognized in the computer vision research community and largescale methods have become an active topic of research in recent years, see among others [1,2,3,4,5,6,7,8,9,10].

The introduction of the ImageNet dataset [1], which contains more than 14M manually labeled images of 22K classes, has provided an important benchmark for large-scale image classification and annotation algorithms.

**Comment by sweaterr** [comment]
A drawback of these methods, however, is that when images of new categories become available, new classifiers have to be trained at a relatively high computational cost.

**Comment by sweaterr** [comment]
Many real-life large-scale datasets are open-ended and dynamic: new classes appear over time and new photos/videos are continuously added to existing classes.

Therefore, the main objective of our work is to propose and study approaches which enable the addition of new classes and new images to existing classes at (near) zero cost.

We explore two techniques to do so.

The first is k-NN classification, where images of new classes are added in the database, and can be used for classification without further processing.

This is a highly non-linear and non-parametric classifier that has shown competitive performance for image annotation [3,7,12].

Its main drawback is that the nearest neighbor search for classification is computationally demanding for large and high-dimensional datasets.

The second is the nearest class mean classifier (NCM), where classes are represented by their mean feature vector [13,14,15].

The cost of computing the mean can be neglected with respect to the cost of feature extraction and this operation does not require accessing images of other classes.

Contrary to the k-NN classifier, this is a linear classifier which leads to efficient classification.

The complete distribution of the training data of a class is, however, only characterized by its mean and it is therefore unclear whether this is sufficient for competitive performance.

**Comment by sweaterr** [comment]
The success of both methods critically depends on the metric which is used to compute the distance between an image and other images (for k-NN) or class means (for NCM), see Figure 1 for an illustration.

Metric learning has received much attention in the machine learning and computer vision communities recently, and has been shown to significantly increase the performance of distance-based classifiers.

Therefore, we cast our classifier learning problem as one of learning a metric which is shared across all classes.

For k-NN classification, we use the Large Margin Nearest Neighbor (LMNN) framework [16] and propose a variation that significantly improves its performance.

For the NCM classifier, we propose a novel metric learning algorithm based on multi-class logistic discrimination.

A sample from a given class is enforced to be closer to its class mean than to any other class mean in the projected space.

To apply these metric learning techniques on large-scale datasets, we employ stochastic gradient descend (SGD) algorithms, which access only a small fraction of the training data at each iteration [17].

To allow metric learning on high-dimensional image features of large scale datasets that are too large to fit in memory, we in addition use product quantization [18], a data compression technique that was recently used with success for large-scale image retrieval [6] and classifier training [8].

**Comment by sweaterr** [comment]
We report experiments on the ImageNet Large Scale Visual Recognition Challenge 2010 (ILSVRC’10) dataset, which consists of 1.2M train images of 1,000 classes.

As a baseline approach, we use the setup of the winning entry in the 2011 edition of the challenge [8]: Fisher vector image representations [19] are used to describe images and one-vs-rest linear SVM classifiers are learned independently for each class.

Surprisingly, we find that the NCM classifier outperforms the more flexible k-NN classifier.

Moreover, the NCM classifier performs on par with the SVM baseline.

We also consider the generalization performance to new classes.

In a first experiment, we train the metric on a subset of classes of ILSVRC’10 and include the held-out classes at test time.

We only observe a small drop in performance compared to the experiment where the metric is learned with all classes.

In a second experiment, we train the metric on ILSVRC’10 and apply it to a larger set of 10K ImageNet classes.

Once the metric is  learned, we can learn the 10K classifiers (i.e, compute their means) on 64K dimensional features in less than an hour on a single CPU, while learning one-vs-rest linear SVMs on the same data takes on the order of 280 CPU days.

Finally, we explore a zero-shot setting where the class mean of novel classes are estimated based on related classes in the ImageNet hierarchy.

We show that the zero-shot class mean can be effectively combined with the empirical mean of a small number of training images.

This provides an approach that smoothly transitions from settings without training data to ones with abundant training data.

The rest of the paper is organized as follows.

In Section 2 we discuss a selection of related work which is most relevant to this paper.

In Section 3 we present the metric learning techniques we explored for the k-NN and NCM classifiers.

We then present extensive experimental results in Section 4, followed by our conclusions in Section 5.
